src/errorpage.cc

```diff
@@ -1348,15 +1348,12 @@ ErrorState::BuildHttpReply()
     // Make sure error codes get back to the client side for logging and
     // error tracking.
     if (request) {
-        if (detail)
-            request->detailError(type, detail);
-        else
-            request->detailError(type, SysErrorDetail::NewIfAny(xerrno));
+        request->error.update(type, detail);
+        request->error.update(SysErrorDetail::NewIfAny(xerrno));
     } else if (ale) {
-        if (detail)
-            ale->updateError(Error(type, detail));
-        else
-            ale->updateError(Error(type, SysErrorDetail::NewIfAny(xerrno)));
+        Error err(type, detail);
+        err.update(SysErrorDetail::NewIfAny(xerrno));
+        ale->updateError(err);
```

Comment:
Unlike HttpRequest, ALE does not expose its error storage, so we cannot, without heroic efforts, keep this code symmetric to the more efficient HttpRequest code above. I suspect we will eventually adjust ALE to expose its error, but that requires a few noisy changes and should probably be done _after_ ErrorState refactoring mentioned in another change request comment. For now, I think this lack of symmetry is acceptable in this (already problematic) code.